src/components/PeopleTable.tsx

```diff
@@ -1,644 +1,141 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import React, { useEffect, useState } from 'react';
+import { CompletePerson, Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { SearchParams } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  peopleList: Person[];
+  searchParams: URLSearchParams;
+};
+
+enum SortableKeys {
+  name = 'name',
+  sex = 'sex',
+  born = 'born',
+  died = 'died',
+}
+
+export const PeopleTable: React.FC<Props> = ({ peopleList, searchParams }) => {
+  const [newList, setNewList] = useState<CompletePerson[]>([]);
+  const [className, setClassName] = useState('fas fa-sort');
+
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const sortableKeys = Object.keys(SortableKeys);
+
+  const findParent = (parentName: string | null) => {
+    return parentName
+      ? peopleList.find(person => person.name === parentName) || null
+      : null;
+  };
+
+  const newPeopleList = peopleList.map(person => ({
+    ...person,
+    mother: findParent(person.motherName),
+    father: findParent(person.fatherName),
+  }));
+
+  useEffect(() => {
+    setNewList(newPeopleList);
+  }, [peopleList]);
```

Comment:
Consider adding `newPeopleList` to the dependency array of this `useEffect` to ensure it updates correctly when `peopleList` changes.